tests/ui_tests/test_profile_page.py

```diff
@@ -0,0 +1,85 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import allure
+import pytest
+
+from adcm_pytest_plugin.utils import wait_until_step_succeeds
+
+from tests.ui_tests.app.app import ADCMTest
+from tests.ui_tests.app.page.admin_intro.page import AdminIntroPage
+from tests.ui_tests.app.page.login.page import LoginPage
+from tests.ui_tests.app.page.profile.page import ProfilePage
+from tests.ui_tests.utils import restore_admin_password
+
+# pylint: disable=redefined-outer-name
+
+
+@pytest.mark.usefixtures("login_to_adcm_over_api")
+def test_open_profile(app_fs: ADCMTest):
+    """
+    Open profile page via UI elements, check username is correct and required fields presented
+    """
+    params = {'username': 'admin'}
+    intro_page = AdminIntroPage(app_fs.driver, app_fs.adcm.url)
+    intro_page.header.open_profile()
+    profile_page = ProfilePage(intro_page.driver, intro_page.base_url)
+    profile_page.wait_page_is_opened()
+    profile_page.check_required_fields_are_presented()
+    _assert_username_on_profile_page(profile_page, params['username'])
+
+
+@pytest.mark.usefixtures("login_to_adcm_over_api")
+def test_login_as_new_user(another_user: dict, app_fs: ADCMTest):
+    """Login as admin, logout, login as another user, check username"""
+    params = {'admin_username': 'admin', 'another_username': another_user['username']}
+    profile_page = ProfilePage(app_fs.driver, app_fs.adcm.url).open()
+    _assert_username_on_profile_page(profile_page, params['admin_username'])
+    profile_page.header.logout()
+    login_page = LoginPage(profile_page.driver, profile_page.base_url)
+    login_page.wait_page_is_opened()
+    login_page.login_user(**another_user)
+    intro_page = AdminIntroPage(profile_page.driver, profile_page.base_url)
+    intro_page.wait_config_loaded()
+    intro_page.header.open_profile()
+    profile_page.wait_page_is_opened()
+    profile_page.check_required_fields_are_presented()
+    _assert_username_on_profile_page(profile_page, another_user['username'])
+
+
+@pytest.mark.usefixtures("login_to_adcm_over_api")
+def test_change_password(adcm_credentials: dict, app_fs: ADCMTest):
+    """Change admin password over UI and login under new credentials"""
+    new_password = 'password'
+    profile_page = ProfilePage(app_fs.driver, app_fs.adcm.url).open()
+    with restore_admin_password(new_password, adcm_credentials, app_fs.adcm.url) as new_credentials:
```

Comment:
I think there is no need to change admin password back. This is a test of changing the password from the default to a new one and it shouldn't contain any additional hidden steps

Reply:
@vgorkavenko During teardown ADCM fixture removes hosts. That action requires authentication. As we changed the admin password in test, we need to set it back. I guess, it'll be better to make a fixture that sets the password back again, but not sure about order of teardown calls.
Maybe smt like code below will work?
```python
@pytest.fixture(params=[{'username': 'admin', 'password': 'newpass'}])
def restore_password(request, adcm_credentials, adcm_fs):
    new_creds = request.param
    ...
```

Reply:
Reworked as mentioned above

Reply:
Do we create real hosts in some UI tests?

Reply:
Nope, its just default implementation of a teardown in `_adcm` method that is used in `adcm_*` fixtures

Reply:
I think it needs not to create a separate fixture because of this and somehow handle this situation in tests in which we do not create real hosts. @dgusakov ?

Reply:
@Sealwing This call https://github.com/arenadata/adcm_pytest_plugin/blob/a2b5595e1af39c2384c0628d2b985b83e6e9d5fe/src/adcm_pytest_plugin/fixtures.py#L189 should be removed since there is other places where we should remove hosts. After step removal from adcm_pytest_plugin credentials restores will become useless here.